apps/server/src/utils/InstitutionResolver.ts

```diff
@@ -0,0 +1,50 @@
+import type { InstitutionProvider, LocalInstitution, Provider, ResolvedInstitution } from '../shared/contract'
+import ElasticsearchClient from './elasticSearchClient'
+
+export async function resolveInstitutionProvider (institutionId: string): Promise<ResolvedInstitution> {
+  const institution = await ElasticsearchClient.getInstitution(institutionId)
+  const providers = availableProviders(institution)
+
+  if (providers.length === 1) {
+    let provider = providers[0]
+    const institutionProvider = institution[provider as keyof LocalInstitution] as InstitutionProvider
+    if (provider === 'mx') {
+      if (institution.is_test_bank) {
+        provider = 'mx_int'
+      }
+    }
+    return {
+      id: institutionProvider.id,
+      url: institution.url,
+      name: institution.name,
+      logo_url: institution.logo,
+      provider: provider as Provider
+    }
+  } else {
+    // default to MX for now until further logic is added
```

Comment:
Do we not have any institutions that mx isn't a provider for currently?

Reply:
I think we should write this as if there are institutions that we don't support in our list. We can make a defaultProvider const in this file that we can fix when we do preferences, but this needs to handle if the defaultProvider isn't in the list of providers.

Reply:
All the institutions are from MX's database so MX is a provider for every one. I assumed everything in here would change drastically, I just wanted to create something that did the bare minimum to unblock myself. Now that tests are passing I would prefer to merge this and modify the institutionResolver in a separate MR and ticket for creating the resolver. What do you think?

Reply:
I'm working on the resolver right now, so we can leave this super simple